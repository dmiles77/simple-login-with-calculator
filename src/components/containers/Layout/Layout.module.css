@import '../../../variables.css';

.LayoutStyles {
    width: 100%;
    height: 100%;
    background-color: var(--background-main-light-theme);
}