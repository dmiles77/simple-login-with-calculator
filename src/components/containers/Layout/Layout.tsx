import React from 'react';
import styles from './Layout.module.css';

interface IProps {}

const Layout: React.FC<IProps> = ({ children }) => {
	const { LayoutStyles } = styles;

	return (
		<div className={LayoutStyles}>
			<main>{children}</main>
		</div>
	);
};

export default Layout;
