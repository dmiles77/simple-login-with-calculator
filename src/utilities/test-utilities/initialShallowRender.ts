import { ShallowWrapper } from 'enzyme';

// Used to quickly test basic shallow wrapper components for render success:
const initialShallowRender = (
	comp: ShallowWrapper,
	className?: string
): void => {
	expect(comp.length).toBe(1);
	expect(comp.length).not.toBe(2);
	if (className) expect(comp).toHaveClassName(`${className}`);
};

export default initialShallowRender;
