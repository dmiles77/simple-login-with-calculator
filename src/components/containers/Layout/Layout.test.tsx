import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Layout from './Layout';
import initialShallowRender from '../../../utilities/test-utilities/initialShallowRender';

describe('Layout component', () => {
	let component: ShallowWrapper;

	beforeEach(() => (component = shallow(<Layout />)));

	it('should render without errors', () =>
		initialShallowRender(component, '.LayoutStyles'));

	it('should render a main tag that contains the children prop', () => {
		const main = component.find('main');

		expect(main.length).toBe(1);
		expect(main.children()).toBeTruthy();
	});
});
