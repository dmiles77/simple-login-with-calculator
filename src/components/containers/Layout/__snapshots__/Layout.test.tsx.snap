// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should render without errors 1`] = `
<div
  className="LayoutStyles"
>
  <main />
</div>
`;
